django/text_extractor/templates/text_extractor/completed_documents.html

```diff
@@ -82,10 +82,13 @@
   <script>document.querySelector("#download-all-files").classList.remove("d-none");</script>
 {% endif %}
 {% if user_request_id %}
-  {# Push URL to history for this specific user request #}
-  <script>history.replaceState(null, '', "{% url 'text_extractor:poll_tasks' user_request_id=user_request_id %}");</script>
-{% endif %}
-{% if refresh_on_load %}
-  {# Refresh the page immediately #}
-  <script>window.location.reload();</script>
+  <script>window.location.href = "{% url 'text_extractor:poll_tasks' user_request_id=user_request_id %}";</script>
 {% endif %}
+<script>
```

Comment:
Currently, submitting a new document after processing a previous one takes you to a new page, with your old docs not listed. You can return to them with the Back button (or by using the hash URL if you somehow remembered it), but maybe it's worth having more clear navigation options for users to return to prevoiusly-processed files?

Reply:
Yeah, this was a compromise required for the Celery task polling. My hope is that the back navigation is clear enough.